test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +95,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (c.isMutating()) {
+            assertEquals(addressBook, saveFile.load());
+        } else assertNotSame(addressBook, saveFile.load());
```

Comment:
This shouldn't happen since both your addressBook and saveFile shouldn't have changed if it's not mutating,
also please note that `assertNotSame` isn't the same as `!assertNotEquals`. `assertNotSame` is asserting whether they are same object, whereas `assertEquals` is checking if they have they have the same values.

Also not that your `else` statement should have curly brackets as your `if` statement to make it neat and standardize